composer.json

```diff
@@ -10,22 +10,35 @@
         {"name": "Roman Borschel", "email": "roman@code-factory.org"},
         {"name": "Benjamin Eberlei", "email": "kontakt@beberlei.de"},
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"},
-        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"}
+        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"},
+        {"name": "Fabio B. Silva", "email": "fabio.bat.silva@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
-        "doctrine/lexer": "1.*"
+        "php":          ">=7.0.0",
+        "hoa/compiler": "3.*"
     },
     "require-dev": {
-        "doctrine/cache": "1.*",
-        "phpunit/phpunit": "4.*"
+        "doctrine/cache"                        : "1.*",
+        "phpunit/phpunit"                       : "5.*",
+        "instaclick/coding-standard"            : "~1.1",
+        "squizlabs/php_codesniffer"             : "~1.5",
+        "satooshi/php-coveralls"                : "~1.0",
+        "instaclick/object-calisthenics-sniffs" : "dev-master",
+        "instaclick/symfony2-coding-standard"   : "dev-remaster"
     },
     "autoload": {
-        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
+        "psr-4": {
+            "Doctrine\\Annotations\\": "src/"
```

Comment:
@mickaelandrieu the symfony/symfony package depends on doctrine/annotations (well, on ``doctrine/common`` for now but we should switch to more specific components). So any project wanting to use symfony will also have to use a version compatible with the symfony requirement. This is what I'm telling here. It is just basic of how dependency resolution works.
And this is not specific to Symfony. It is true for *any package depending on your library*. I took the example of Symfony here, because it is a package with a big impact. I could have taken any of the 359 packages listed on https://packagist.org/packages/doctrine/annotations/dependents (btw, Zend Framework is also impacted), but https://packagist.org/packages/1of0/better-annotations would not have been as good as example as it only have 9 installs...

If you want to make it very hard to upgrade to doctrine/annotations 1.x, fine for you, but be aware that this means forcing a hard split of the packages relying on your library, between packages wanting 1.x and packages wanting 2.x (which cannot be installed at the same time), as they won't be able to easily accept both.
And the **only** thing forcing this hard split is the namespace change, as said previously. Without it, most packages will be able to support both 1.x and 2.x without changing any line of their code. This would avoid splitting the community of dependants.
There is no benefit for users in the class renaming, but there is a huge cost. And btw, there is also a maintenance cost as any library not able to upgrade to 2.x only will ask you to continue maintaining your 1.x version in parallel for a long time. If supporting both versions was easy, it would be acceptable to drop support for 1.x after some time (hopefully not as soon as 2.x is released though, but for instance at the end of the official support of PHP 5.6), as anyone running their project on PHP 7 would already be using the 2.x maintained version.

Majors versions are indeed the time where BC breaks are allowed, but each BC break are a cost on the upgrade path (and on maintenance of the library due to the slower adoption of hard upgrades). So breaking BC just because you can, without an actual benefit, is not a good idea. A good read about that are the rules set by the PHP team for the PHP 7 version. They avoided any BC break which would cause more pains than what it was worth (so no signature change for needle/haystack vs haystack/needle for instance), precisely to make adoption of PHP 7 easier. This is also well explained by the Facebook team, which recognize this points in their official policy for React: http://facebook.github.io/react/blog/2016/02/19/new-versioning-scheme.html#avoiding-the-major-cliff (there is nothing specific to React or NPM in the problem here, you could substitute any package name and any semver-based tool to apply it elsewhere, and this is actually also the policy applied by Symfony for Symfony 3)